src/session/session_helper.c

```diff
@@ -134,3 +134,35 @@ __wt_session_dump(WT_SESSION_IMPL *session, WT_SESSION_IMPL *dump_session, bool
     __wt_scr_free(session, &buf);
     return (ret);
 }
+
+/*
+ * __wt_session_set_last_error --
+ *     Stores information about the last error to occur during this session.
+ */
+int
+__wt_session_set_last_error(
+  WT_SESSION *wt_session, int err, int sub_level_err, const char *err_msg_content)
+{
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session = (WT_SESSION_IMPL *)wt_session;
+
+    char *err_msg = session->err_info.err_msg;
+    size_t err_msg_size;
+
+    /* Ensure arguments are valid, and that session is not a mock */
+    WT_ASSERT(session, __wt_is_valid_sub_level_error(sub_level_err));
+    if (err_msg_content == NULL || wt_session->get_last_error == NULL)
+        return (0);
+
+    /* Free the last error message string, if it was allocated. */
+    __wt_free(session, err_msg);
```

Comment:
IMO, we should have a safeguard if statement for this.

Reply:
What would that look like? What else do I need to check?